src/render.rs

```diff
@@ -108,8 +101,7 @@ impl<'a> Renderer<'a> {
 
         // dummy statement to satisfy the compiler
         // TODO: make it so that's not needed
-        self.context.pointer(&self.get_json_pointer(key)).cloned()
-            .ok_or_else(|| FieldNotFound(key.to_string()))
+        self.context.pointer(&get_json_pointer(key)).cloned().ok_or_else(|| FieldNotFound(key.to_string()))
```

Comment:
Same thing here, though it's not as bad.

Also, why not use `unreachable!` here?